src/app/workflow/register-workflow-modal/register-workflow-modal.component.ts

```diff
@@ -237,6 +238,23 @@ export class RegisterWorkflowModalComponent implements OnInit, AfterViewChecked,
     this.descriptorValidationPattern = DescriptorLanguageService.workflowDescriptorTypeEnumToExtendedDescriptorLanguageBean(
       descriptorType
     ).descriptorPathPattern;
+    switch (descriptorType) {
+      case Workflow.DescriptorTypeEnum.CWL:
+        this.workflowPathError = validationMessages.cwlPath.pattern;
+        break;
+      case Workflow.DescriptorTypeEnum.WDL:
+        this.workflowPathError = validationMessages.wdlPath.pattern;
+        break;
+      case Workflow.DescriptorTypeEnum.NFL:
+        this.workflowPathError = validationMessages.nflPath.pattern;
+        break;
+      case Workflow.DescriptorTypeEnum.Gxformat2:
+        this.workflowPathError = validationMessages.galaxyPath.pattern;
+        break;
+      default:
```

Comment:
I don't think you should have a default. That way the compiler can catch a missing one.

OTOH, we're a long way from turning on strictnullchecks. And if we don't have it, then maybe default is OK.

Will let others call it.

Reply:
Wasn't aware that the compiler could catch the missing one

Reply:
On further reflection, go with default. We're too far away from having strictnullchecks.

FYI, see [first answer](https://stackoverflow.com/questions/39419170/how-do-i-check-that-a-switch-block-is-exhaustive-in-typescript) for how compiler could catch.

Reply:
ok, i interpreted your comment differently I guess.  Thought you meant just removing default when in fact you need to pull it to a different function and have strictnullcheck ensure a non-null response from it.